File: src/main/java/mil/army/futures/asitemplate/services/ProductService.java
package mil.army.futures.asitemplate.services;

import lombok.AllArgsConstructor;
import mil.army.futures.asitemplate.Product;
import mil.army.futures.asitemplate.repositories.ProductRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class ProductService {

    ProductRepository productRepository;

    public Product addProduct(Product product) {
        return productRepository.save(product);
    }

    public List<Product> getProducts() {
        return productRepository.findAll();
    }
}

File: src/main/kotlin/mil/army/futures/asitemplate/services/ProductService.kt
package mil.army.futures.asitemplate.services

import mil.army.futures.asitemplate.Product
import mil.army.futures.asitemplate.repositories.ProductRepository
import org.springframework.stereotype.Service

@Service
class ProductService(private val productRepository: ProductRepository) {
    fun addProduct(product: String): Product {
        return productRepository.save(Product(name = product, quantity = 0))
    }

    fun getProducts(): List<Product> {
        return productRepository.findAll()
    }
}